crates/bevy_ecs/src/system/builder.rs

```diff
@@ -382,4 +411,33 @@ mod tests {
         let result = world.run_system_once(system);
         assert_eq!(result, 5);
     }
+
+    #[test]
+    fn dyn_builder() {
```

Comment:
I would like another test verifying that the unhappy path panics.

Reply:
I've got the `assert!(p0.downcast_mut::<Query<()>>().is_none());` there to verify that you can't downcast to the wrong type.  (It returns `Option`, so there doesn't need to be a panic.)  Do you want that moved to a separate test?

Reply:
Hmm, it can stay where it is I suppose.